src/dumbo/src/ns.rs

```diff
@@ -104,32 +109,40 @@ impl MmdsNetworkStack {
     // This is the entry point into the MMDS network stack. The src slice should hold the contents
     // of an Ethernet frame (of that exact size, without the CRC).
     pub fn detour_frame(&mut self, src: &[u8]) -> bool {
-        // The frame cannot possibly contain an ARP request or IPv4 packet for the MMDS.
-        if !test_speculative_tpa(src, self.ipv4_addr)
-            && !test_speculative_dst_addr(src, self.ipv4_addr)
-        {
+        let mut arp_reply_ipv4_addr = None;
+        for ipv4_addr in self.ipv4_addr_pool.iter() {
+            // Check if the frame contains an ARP frame or an IPv4 packet sent to one
+            // of the MMDS server IPv4 addresses.
+            if test_speculative_tpa(src, *ipv4_addr) || test_speculative_dst_addr(src, *ipv4_addr) {
+                arp_reply_ipv4_addr = Some(*ipv4_addr);
+            }
+        }
```

Comment:
Question: Do we want here to keep only the last ipv4 address that meets one of these conditions? If yes, could we iterate the `Vec` in reverse order and add a `break` if we found a "good" `ipv4_addr`?